internal/controller/setup.go

```diff
@@ -92,6 +93,7 @@ func SetupControllers(logger logr.Logger, mgr manager.Manager, options SetupOpti
 	}
 
 	if versionInfo != nil {
+		mf.OperatorMetadata.KubernetesVersion = versionInfo.String()
```

Comment:
maybe we could define `discoveryClient` one level up and pass it to `SetupControllers`, instead of passing `mf`? (since `mf` isn't related to controller setup.) then we could declare `versionInfo` one level up too and pass it to `setupMetadataForwarder`